packages/botonic-plugin-flow-builder/src/content-fields/flow-carousel.tsx

```diff
@@ -23,7 +28,77 @@ export class FlowCarousel extends ContentFieldsBase {
     return newCarousel
   }
 
-  toBotonic(id: string): JSX.Element {
+  static fromAIAgent(
+    id: string,
+    carouselMessage: CarouselMessage,
+    request: ActionRequest
+  ): JSX.Element {
+    const canUseWhatsappInteractiveMediaCarousel = () => {
+      const isValid =
+        carouselMessage.content.text &&
+        !carouselMessage.content.elements.some(
+          element => element.button.payload
+        ) &&
+        carouselMessage.content.elements.every(element => element.button.url)
+
+      if (!isValid) {
+        console.warn(
+          'Cannot use WhatsappInteractiveMediaCarousel for Whatsapp created by AIAgent',
+          carouselMessage.content
+        )
+      }
+      return isValid
+    }
+
+    if (
+      isWhatsapp(request.session) &&
+      canUseWhatsappInteractiveMediaCarousel()
+    ) {
+      return (
+        <WhatsappInteractiveMediaCarousel
+          cards={carouselMessage.content.elements.map(element => {
+            const buttonText = element.button.text
+            const buttonUrl = element.button.url!
+            const imageLink = element.image
+
+            return {
+              text: element.title,
+              action: { buttonText, buttonUrl, imageLink },
+            }
+          })}
+          textMessage={carouselMessage.content.text!}
```

Comment:
i would place the textMessage first when declaring it as I understand it will be the first thing being rendered